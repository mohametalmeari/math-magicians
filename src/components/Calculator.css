
.container {
  display: flex;
  flex-direction: column;
  max-width: 50%;
}

.calc-pad {
  display: grid;
  grid-template-columns: auto auto auto auto;
  aspect-ratio: 3/2;
  border: solid gray 1px;
}

.calc-pad * {
  border: solid gray 1px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: white;
}

.operator {
  background-color: orangered;
}

.num0 {
  grid-column: 1/ span 2;
}

.result {
  display: flex;
  align-items: stretch;
  justify-content: end;
  background: gray;
  color: white;
  padding: 1rem 0.5rem;
  margin: 0;
  font-size: x-large;
}
