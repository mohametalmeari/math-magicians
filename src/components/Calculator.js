import { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [result, setResult] = useState({});
  const calc = (e) => {
    setResult(calculate(result, e.target.value));
  };
  return (
    <div className="calculator-container">
      <p className="calculator-text">Let&apos;s do some math!</p>
      <div className="calculator">
        <p className="result">
          {result.next ? result.next : result.total}
        </p>
        <div className="calc-pad" role="presentation" onClick={(e) => { calc(e); }}>
          <button type="button" className="AC" value="AC">AC</button>
          <button type="button" className="sign" value="+/-">+/-</button>
          <button type="button" className="percent" value="%">%</button>
          <button type="button" className="operator division" value="รท">รท</button>
          <button type="button" className="num num7" value="7">7</button>
          <button type="button" className="num num8" value="8">8</button>
          <button type="button" className="num num9" value="9">9</button>
          <button type="button" className="operator multiplication" value="x">ร</button>
          <button type="button" className="num num4" value="4">4</button>
          <button type="button" className="num num5" value="5">5</button>
          <button type="button" className="num num6" value="6">6</button>
          <button type="button" className="operator subtraction" value="-">-</button>
          <button type="button" className="num num1" value="1">1</button>
          <button type="button" className="num num2" value="2">2</button>
          <button type="button" className="num num3" value="3">3</button>
          <button type="button" className="operator Addition" value="+">+</button>
          <button type="button" className="num num0" value="0">0</button>
          <button type="button" className="num decimal" value=".">.</button>
          <button type="button" className="operator equal" value="=">=</button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
