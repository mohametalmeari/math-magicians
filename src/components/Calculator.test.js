import { render } from '@testing-library/react';
import Calculator from './Calculator';

describe('Calculator Component', () => {
  test('test render of calculator page', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
