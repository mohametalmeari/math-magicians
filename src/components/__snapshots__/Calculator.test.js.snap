// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component renders without errors 1`] = `
<div>
  <div
    class="calculator-container"
  >
    <p
      class="calculator-text"
    >
      Let's do some math!
    </p>
    <div
      class="calculator"
    >
      <p
        class="result"
      />
      <div
        class="calc-pad"
        role="presentation"
      >
        <button
          class="AC"
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          class="sign"
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          class="percent"
          type="button"
          value="%"
        >
          %
        </button>
        <button
          class="operator division"
          type="button"
          value="÷"
        >
          ÷
        </button>
        <button
          class="num num7"
          type="button"
          value="7"
        >
          7
        </button>
        <button
          class="num num8"
          type="button"
          value="8"
        >
          8
        </button>
        <button
          class="num num9"
          type="button"
          value="9"
        >
          9
        </button>
        <button
          class="operator multiplication"
          type="button"
          value="x"
        >
          ×
        </button>
        <button
          class="num num4"
          type="button"
          value="4"
        >
          4
        </button>
        <button
          class="num num5"
          type="button"
          value="5"
        >
          5
        </button>
        <button
          class="num num6"
          type="button"
          value="6"
        >
          6
        </button>
        <button
          class="operator subtraction"
          type="button"
          value="-"
        >
          -
        </button>
        <button
          class="num num1"
          type="button"
          value="1"
        >
          1
        </button>
        <button
          class="num num2"
          type="button"
          value="2"
        >
          2
        </button>
        <button
          class="num num3"
          type="button"
          value="3"
        >
          3
        </button>
        <button
          class="operator Addition"
          type="button"
          value="+"
        >
          +
        </button>
        <button
          class="num num0"
          type="button"
          value="0"
        >
          0
        </button>
        <button
          class="num decimal"
          type="button"
          value="."
        >
          .
        </button>
        <button
          class="operator equal"
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Calculator Component test render of calculator page 1`] = `
<div>
  <div
    class="calculator-container"
  >
    <p
      class="calculator-text"
    >
      Let's do some math!
    </p>
    <div
      class="calculator"
    >
      <p
        class="result"
      />
      <div
        class="calc-pad"
        role="presentation"
      >
        <button
          class="AC"
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          class="sign"
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          class="percent"
          type="button"
          value="%"
        >
          %
        </button>
        <button
          class="operator division"
          type="button"
          value="÷"
        >
          ÷
        </button>
        <button
          class="num num7"
          type="button"
          value="7"
        >
          7
        </button>
        <button
          class="num num8"
          type="button"
          value="8"
        >
          8
        </button>
        <button
          class="num num9"
          type="button"
          value="9"
        >
          9
        </button>
        <button
          class="operator multiplication"
          type="button"
          value="x"
        >
          ×
        </button>
        <button
          class="num num4"
          type="button"
          value="4"
        >
          4
        </button>
        <button
          class="num num5"
          type="button"
          value="5"
        >
          5
        </button>
        <button
          class="num num6"
          type="button"
          value="6"
        >
          6
        </button>
        <button
          class="operator subtraction"
          type="button"
          value="-"
        >
          -
        </button>
        <button
          class="num num1"
          type="button"
          value="1"
        >
          1
        </button>
        <button
          class="num num2"
          type="button"
          value="2"
        >
          2
        </button>
        <button
          class="num num3"
          type="button"
          value="3"
        >
          3
        </button>
        <button
          class="operator Addition"
          type="button"
          value="+"
        >
          +
        </button>
        <button
          class="num num0"
          type="button"
          value="0"
        >
          0
        </button>
        <button
          class="num decimal"
          type="button"
          value="."
        >
          .
        </button>
        <button
          class="operator equal"
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
